src/app-layer-modbus.c

```diff
@@ -612,6 +612,11 @@ static void ModbusParseReadResponse(ModbusTransaction   *tx,
             goto error;
     }
 
+    tx->data = SCMalloc(count);
```

Comment:
This looks suspicious, and even if it isn't it should have some comments. 2 things that I think of right away:
1. "can count be 0 or large?" Seems not possible but might be good add a comment stating it explicitly.
2. "can tx->data already be set?" If so it's overwritten. If not, well that is a good thing to mention in a comment as well.